pandas/tests/reshape/test_get_dummies.py

```diff
@@ -45,6 +45,12 @@ def test_get_dummies_raises_on_dtype_object(self, df):
         with pytest.raises(ValueError, match=msg):
             get_dummies(df, dtype="object")
 
+    def test_get_dummies_warns_default_dtype(self, df):
+        # https://github.com/pandas-dev/pandas/issues/45848
+        msg = "In a future version of pandas the default dtype will change"
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            get_dummies(df)
```

Comment:
sorry to go back on the approval, but following Will's comment - can we check the return value here?